### PR TITLE
Fei homework04, not finished, stuck at the move assignment...

### PR DESCRIPTION
**Pull request by feicui08**
cannot build because of the error on move assignment. 
expected ',' or '...' before '&&' token.
'src ' was not declared in this scope.

### CODE REVIEW

**Comment by vargad** [comment]
It builds for me, are you sure it uses the newer comiler? You can check by running ```make VERBOSE=1``` in the build directory, the gcc/g++ it runs should be at least gcc 4.9, and it has to pass -std=c++14 flag to the compiler.

File: course5/homework04/homework04.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    Stream& operator <<(FlushEndl)
+    {
+        std::fprintf(output_file, "%s", "\n");
+        std::fflush(output_file);
+        return *this;
+    }
+
+    Stream& operator =(Stream &&src)
```
Comment (maciekgajewski) [inline course5/homework04/homework04.cc:52]: you'd also need a move constructor

File: course5/homework04/homework04.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+
+Stream Cout("/dev/stdout");
+FlushEndl Endl;
+
+
+int main()
+{
+    Stream s("output.txt");
```
Comment (maciekgajewski) [inline course5/homework04/homework04.cc:70]: test move constructor, test move assignment

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
Add move constructor
Test move constructor and move assignment

**Pull request closed without merging**